### ISSUE TITLE
Feature: support filtering on dateUpdated

### ISSUE THREAD

**Issue by jin-harmoney** [GoogleCloudPlatform/gcr-cleaner#149]
### TL;DR

Even though images might be old (created a long time ago), they might have been updated recently, so it would be handy to make the difference between `dateCreated` vs `dateUpdated` when checking the `grace` period.

### Proposed behavior

Introduce a new flag: `useDateUpdated`.

When setting the flag `useDateUpdated` to `true`, images are filtered against `dateUpdated` instead of `dateCreated` (or `dateUploaded` as mentioned in the code):

https://github.com/GoogleCloudPlatform/gcr-cleaner/blob/main/pkg/gcrcleaner/cleaner.go#L335

![image](https://github.com/GoogleCloudPlatform/gcr-cleaner/assets/73221968/7c529f7b-8ac0-47d7-a534-ad54fec13a4d)

### How are you running gcr-cleaner?

CLI

### gcr-cleaner version

Latest

### Environment

Using Cloud Run and Cloud Scheduler on GCP.

**Comment by sethvargo** [comment]
Hi all - users should prefer the native Google Artifact Registry functionality instead of gcr-cleaner. We are only fixing bugs and security issues in gcr-cleaner now that there's a native (and free) feature in the Google Cloud product.
